juspay/hyperswitch
# feat(core): create mandates with payment_token

## Type of Change



## Description
This PR supports creating a mandate by passing `payment_token` with , `setup_future_usage:off_session` and `customer_acceptance`


### Additional Changes





## Motivation and Context



## How did you test it?
- Create an MA and MCA
- Make a  save card payment
- List Customer Payment Method, get the `payment_token`
- Use the `payment token` to create the mandate 
> Create
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_1ORGV1egb2qN8z8UoZE7pCkv7cUMiXbdMwZYZJNX8qjm2aTW9uAcX2d3ezX4g3KY' \
--data-raw '{
    "amount": 6777,
    "currency": "USD",
    "confirm": false,
    
    "customer_id": "new-c77",
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "authentication_type": "no_three_ds",
    "return_url": "https://google.com",
     "email": "something@gmail.com",
    "name": "Joseph Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
 
"setup_future_usage":"off_session",
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },

    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "connector_metadata": {
        "noon": {
            "order_category": "pay"
        }
    }
}'
```
>Confirm
```
curl --location 'http://localhost:8080/payments/pay_up5SZAykikpNXZqLivDP/confirm' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_1ORGV1egb2qN8z8UoZE7pCkv7cUMiXbdMwZYZJNX8qjm2aTW9uAcX2d3ezX4g3KY' \
--data '{
    "confirm": true,
          "payment_method": "card",
    "payment_method_type": "credit",
   "payment_token": "{{payment_token}}",
      "customer_acceptance": {
            "acceptance_type": "offline",
            "accepted_at": "1963-05-03T04:07:52.723Z",
            "online": {
                "ip_address": "in sit",
                "user_agent": "amet irure esse"
            }}
}'
```

- Do a recurring payment, with `off_session:true`
>Create 
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_1ORGV1egb2qN8z8UoZE7pCkv7cUMiXbdMwZYZJNX8qjm2aTW9uAcX2d3ezX4g3KY' \
--data-raw '{
    "amount": 1000,
    "currency": "USD",
    "confirm": true,
    
    "customer_id": "new-c77",
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "authentication_type": "no_three_ds",
    "return_url": "https://google.com",
    "email": "something@gmail.com",
    "name": "Joseph Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
       "off_session": true,
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },

    
      "recurring_details":{
        "type":"payment_method_id",
        "data":"{{payment_method_id}}"
    },

    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "connector_metadata": {
        "noon": {
            "order_category": "pay"
        }
    }
}'
```


## Checklist
